src/main/java/apoc/refactor/GraphRefactoring.java

```diff
@@ -396,4 +407,23 @@ private Relationship copyRelationship(Relationship rel, Node source, Node target
         Relationship newrel = startNode.createRelationshipTo(endNode, rel.getType());
         return copyProperties(rel, newrel);
     }
+
+
+
+    public void mergeRelsWithSameTypeAndDirectionInMergeNodes (Node node, RefactorConfig config, Direction dir){
+        List<Relationship> rels = new ArrayList<>();
+        node.getRelationships(dir).iterator().forEachRemaining(rels::add);
+        for (int i = 0; i < rels.size()-1; i++) {
+            for (int j = i + 1; j < rels.size(); j++) {
+                if (rels.get(i).isType(rels.get(j).getType())){
```

Comment:
don't call rels.get() repeatedly, use local variables.
they should have the same start/end-node given that you iterate starting from that one